### PR TITLE
Base convert puzzle

### PR DESCRIPTION
**Pull request by mekealbrown**
-Fixed Solver
-.desc.txt should be mostly good(for this version of the puzzle)
-Removed rock-texture.jpg, just used the actual link to the photo used
-Removed custom footer

### CODE REVIEW

File: src/based-puzzle.cpp [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+      values.push_back(std::to_string(rand));
+      bases.push_back(base);
+    }
+  }
+
+  std::string html_content = utils_html_printf("Base Puzzle", "./files-based/.desc.txt", values);
+  utils_generate_file("./files-based/instructions.html", html_content);
+
+  return Puzzle{"files-based", std::to_string(solve(values, bases))};
```
Comment (malloc-nbytes) [inline src/based-puzzle.cpp:69]: The puzzle struct has one more optional value. If you don't want to put anything there, just put an empty initializer

File: src/based-puzzle.cpp [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        solution |= mask;
+      }
+      mask >>= 1;
+      decimalval = items = 0;
+    }
+  }
+  return solution;
+}
+
+//TODO: Rework the puzzle
```
Comment (malloc-nbytes) [inline src/based-puzzle.cpp:36]: Remove comment if uneeded

File: src/based-puzzle.cpp [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+  std::vector<int> bases;
+
+  for (int i = 0; i < 12; i++) {
+    short rand = utils_rng_roll(0, 67, seed);
+    if (rand > 42) { // Keep them within hex range of values
+      if (rand > 47) {
+        rand = utils_rng_roll(42, 47, seed);
+      }
+
+      values.push_back(std::string(1, static_cast<char>(rand + 23)));
```
Comment (malloc-nbytes) [inline src/based-puzzle.cpp:49]: Could you make `23` be a variable instead of just a magic number

**Review by malloc-nbytes — APPROVED** [review]
LGTM

**Review by TurnerAustin — APPROVED** [review]
LGTM

**Pull request merged**